### ISSUE TITLE
Training accuracy 

### ISSUE THREAD

**Issue by francescamanni1989** [AaronX121/Soft-Decision-Tree#6]
Thanks for your implementation, very insightful!
Anyway, why training accuracy is so low?
![image](https://user-images.githubusercontent.com/83947886/133087307-486dbbc3-ab83-491d-b4ec-7224c7340525.png)
I have been choosing your default hyperparameters and setting.

Thanks!

**Comment by xuyxu** [comment]
I think there should be some problems with your way of calculating the training accuracy, could you show me the code?

**Comment by francescamanni1989** [comment]
yes, ofc.
![image](https://user-images.githubusercontent.com/83947886/133089688-dbfaf260-1b74-402b-8576-8d93bbfe374d.png)

**Comment by xuyxu** [comment]
Please replace the `len(train_loader.dataset)`  in the third line with `data.size(0) `, this should fix your problem.

**Comment by xuyxu** [comment]
len(train_loader.dataset) indicate the total number of batches in the dataset, instead of the number of samples in the current batch.

**Comment by francescamanni1989** [comment]
Thank you! In case I want to implement the F.softmax in the forward step of the model, is it ok to do so after y_pred = self.leaf_nodes(mu)?

**Comment by xuyxu** [comment]
Of course, this should be fine.
